.github/workflows/lint-and-test.yaml

```diff
@@ -3,49 +3,129 @@ name: Lint and test PR
 on: pull_request
 
 jobs:
-  need-test:
+  docker-build:
+    name: Build Docker image
     runs-on: ubuntu-latest
-    outputs:
-      charts: ${{ steps.list-changed.outputs.changed }}
     steps:
       - uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build
+        uses: docker/build-push-action@v4
         with:
-          fetch-depth: 0
-      - uses: helm/chart-testing-action@v2.3.1
-      - id: list-changed
-        run: |
-          changed=$(ct --config .github/ct.yaml list-changed)
-          if [[ -n "$changed" ]]; then
-            echo "changed=true" >> $GITHUB_OUTPUT
-          fi
+          tags: ci.local/${{ github.repository }}:ci
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```

Comment:
So we are compiling always. Any change in the chart, the first iteration when the caches are cold, is going to drag 10 minutes, right?

Reply:
It will be instant (~10s) if it has run before on `main`, or on a PR to `main`. Otherwise, entries stay in the cache for [7 days](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy) since last accessed (not written).

I thought wasting 10 minutes every 7 days (worst case scenario, I would expect renovate prs more often than that) would be an acceptable tradeoff for simpler jobs. But of course, that may be a bit opinionated.